### PR TITLE
add traits quiz

### PR DESCRIPTION
**Pull request by Symmaque**
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/807)

### CODE REVIEW

File: quizzes/ch08-02-traits.toml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    let dog = dog::Dog {};
+    dog.make_noise();
+
+    let cat = cat::Cat {};
+    cat.make_noise();
+}
+"""
+answer.doesCompile = false
+context = """
+This is not possible for trait function to have a body in cairo.
```
Comment (enitrat) [inline quizzes/ch08-02-traits.toml:40]: Actually I think this is coming in the next update, but for now it's indeed valid :)

File: quizzes/ch08-02-traits.toml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    let my_shape = Shape{};
+    // the line goes here
+}
+```
+"""
+prompt.distractors = ["`draw(my_shape);`", "`draw(@my_shape);`"]
+answer.answer = "`my_shape.draw();`"
+context = """
+The `draw` method is a method implemented for the `Shape` type, so it should be called on the `my_shape` object.
+It is also possible to call the method on the snapshot of the object by using the `@` operator : `(@my_shape).draw();` for example.
```
Comment (enitrat) [inline quizzes/ch08-02-traits.toml:73]: yes, it's actually strictly equivalent, basically the compiler transforms my_shape.draw() to (@my_shape).draw()

it's better to say that you can call it with `Drawable::draw(@my_shape)`

Comment (enitrat) [inline quizzes/ch08-02-traits.toml:73]:
```suggestion
It is also possible to call the method on the snapshot of the object by using the `@` operator : `Drawable::draw(@my_shape)` for example.
```

**Review by enitrat — COMMENTED** [review]
great :)

**Review by enitrat — APPROVED** [review]

**Comment by Jemiiah** [comment]
I was still doing the work!!

**Pull request merged**